pennylane/data/dataset.py

```diff
@@ -228,9 +242,13 @@ def write(self, filepath, protocol=4):
         dirname = os.path.dirname(filepath)
         if dirname and not os.path.exists(dirname):
             os.makedirs(dirname)
+        # if some values are still lazy-loaded to None, this will load them before writing
+        if self._is_standard:
+            for attr in self.attrs:
+                _ = getattr(self, attr)
         attrs = self.attrs
         for h in condensed_hamiltonians.intersection(attrs):
-            attrs[h] = hamiltonian_to_dict(getattr(self, h))
+            attrs[h] = self.__hamiltonian_to_dict(attrs[h])
```

Comment:
Why do we use `__` instead of `_` in the method name? Is it some sort of convention?

Reply:
yep, double-underscore methods are private, while single-underscore methods are protected. The difference is not super significant in python, but you have to try a little harder to find private methods from outside the class because the name gets mangled, and I want them hidden!!